front/app/modules/commercial/insights/admin/containers/Insights/Details/Network/index.tsx

```diff
@@ -153,19 +158,35 @@ const Network = ({ params: { viewId } }: WithRouterProps) => {
     } else return true;
   };
 
-  const toggleClusterCollapse = (clusterId: string) => {
-    if (collapsedClusters.includes(clusterId)) {
-      setCollapsedClusters(collapsedClusters.filter((id) => id !== clusterId));
+  const toggleCluster = (node: Node) => {
+    if (collapsedClusters.includes(node.id)) {
+      setCollapsedClusters(collapsedClusters.filter((id) => id !== node.id));
+      forceRef.current?.zoom(2, 400);
+      forceRef.current?.centerAt(node.x, node.y, 400);
     } else {
-      setCollapsedClusters([...collapsedClusters, clusterId]);
+      setCollapsedClusters([...collapsedClusters, node.id]);
     }
   };
-
   const handleNodeClick = (node: Node) => {
-    toggleClusterCollapse(node.id);
-    if (collapsedClusters.includes(node.id)) {
-      forceRef.current?.zoom(2, 400);
-      forceRef.current?.centerAt(node.x, node.y, 400);
+    const isClusterNode = node.cluster_id === null;
+    if (isClusterNode) {
+      toggleCluster(node);
+    } else {
+      clHistory.replace({
+        pathname,
+        search: stringify(
+          // Only add unique keywords to url query
+          {
+            ...query,
+            keywords: query.keywords
```

Comment:
Keeping the keywords in the url as an array complicated things a bit but I still think it's valuable to have shareable filters, especially on this page (View) so I decided to continue with this approach.

Reply:
I think it's a good choice too :) Code is not that complicated in the end either, so well done